### PR TITLE
No release available callback

### PR DESCRIPTION
**Pull request by russelarms**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR adds a new NoReleaseAvailable callback to the Distribute module and fixes the ReleaseAvailable callback on the first start.

## Related PRs or issues

[AB#84056](https://msmobilecenter.visualstudio.com/Mobile-Center/_boards/board/t/Ivan-Team/Backlog%20Items/?workitem=84056)
[AB#84157](https://msmobilecenter.visualstudio.com/Mobile-Center/_boards/board/t/Ivan-Team/Backlog%20Items/?workitem=84157)

##Misc

The listener is initialized after Distribute completes its check, so we probably should start the Distribute module from .Net code.

### CODE REVIEW

File: Assets/AppCenter/Plugins/iOS/Distribute/DistributeDelegate.h [hunk 1]
```diff
@@ -7,11 +7,14 @@
 #import <AppCenterDistribute/AppCenterDistribute.h>
 
 typedef bool (__cdecl *ReleaseAvailableFunction)(MSACReleaseDetails*);
+typedef void (__cdecl *NoReleaseAvailableFunction)();
 
 @interface UnityDistributeDelegate : NSObject<MSACDistributeDelegate>
 
 - (BOOL)distribute:(MSACDistribute *)distribute releaseAvailableWithDetails:(MSACReleaseDetails *)details;
+- (void)noReleaseAvailable;
```
Comment (AnastasiaKubova) [inline Assets/AppCenter/Plugins/iOS/Distribute/DistributeDelegate.h]: This API was renamed to 
`(void)distributeNoReleaseAvailable:(MSACDistribute *)distribute; ` ([PR](https://github.com/microsoft/appcenter-sdk-apple/pull/2238/files#diff-dc19369dffd7a467cc08789d92a9d8217f62b599d3b02d7bf67a8a9497dab0a4R36))

File: Assets/AppCenter/Plugins/AppCenterSDK/Distribute/iOS/DistributeInternal.cs [hunk 2]
```diff
@@ -29,6 +29,22 @@ static bool ReleaseAvailableFunc(IntPtr details)
...
+
+        static NoReleaseAvailableDelegate noReleaseDel;
+        [MonoPInvokeCallback(typeof(NoReleaseAvailableDelegate))]
+        static void NoReleaseAvailableFunc()
+        {
+            if (Distribute.NoReleaseAvailable == null)
+            {
+                return;
+            }
+            Distribute.NoReleaseAvailable.Invoke();
```
Comment (AnastasiaKubova) [inline Assets/AppCenter/Plugins/AppCenterSDK/Distribute/iOS/DistributeInternal.cs]: `Distribute.NoReleaseAvailable?.Invoke();`

File: Assets/AppCenter/Plugins/AppCenterSDK/Distribute/iOS/DistributeInternal.cs [hunk 3]
```diff
@@ -29,6 +29,22 @@ static bool ReleaseAvailableFunc(IntPtr details)
             var releaseDetails = ReleaseDetailsHelper.ReleaseDetailsConvert(details);
             return Distribute.ReleaseAvailable.Invoke(releaseDetails);
         }
+
+#if ENABLE_IL2CPP
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
```
Comment (AnastasiaKubova) [inline Assets/AppCenter/Plugins/AppCenterSDK/Distribute/iOS/DistributeInternal.cs]: Do we really need to duplicate these lines?

File: AppCenterDemoApp/Assets/Puppet/PuppetReleaseHandler.cs [hunk 4]
```diff
@@ -15,6 +15,7 @@ public class PuppetReleaseHandler : MonoBehaviour
     void Awake()
     {
         Distribute.ReleaseAvailable = OnReleaseAvailable;
+        Distribute.NoReleaseAvailable = OnNoReleaseAvailable;
```
Comment (AnastasiaKubova) [inline AppCenterDemoApp/Assets/Puppet/PuppetReleaseHandler.cs:18]: We also have to update changelog

**Review by AnastasiaKubova — APPROVED** [review]

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**